package.json

```diff
@@ -59,26 +66,55 @@
     "ember-disable-prototype-extensions": "^1.1.3",
     "ember-load-initializers": "^2.1.2",
     "ember-resolver": "^8.0.3",
-    "ember-source": "~4.9.1",
+    "ember-source": "~4.8.3",
     "ember-source-channel-url": "^3.0.0",
     "ember-try": "^2.0.0",
     "eslint": "^8.29.0",
     "eslint-config-prettier": "^8.5.0",
     "eslint-plugin-disable-features": "^0.1.3",
     "eslint-plugin-node": "^11.1.0",
     "eslint-plugin-prettier": "^4.2.1",
+    "expect-type": "^0.15.0",
     "loader.js": "^4.7.0",
     "npm-run-all": "^4.1.5",
     "prettier": "2.8.0",
     "qunit": "^2.19.2",
     "release-it": "^15.5.0",
     "release-it-lerna-changelog": "^5.0.0",
+    "typescript": "^4.9.4",
     "webpack": "^5.75.0"
   },
   "peerDependencies": {
     "@ember/test-helpers": "^2.4.0",
+    "@glimmer/interfaces": "^0.84.2",
+    "@glimmer/reference": "^0.84.2",
```

Comment:
@chriskrycho These inclusions seemed to cause CI in `ember-welcome-page` to have failed, after I updated `ember-qunit` to `v6` over there.

https://github.com/ember-cli/ember-welcome-page/actions/runs/3719219299/jobs/6307944375#step:5:142

```sh
ember-qunit has the following unmet peerDependencies:

	* @glimmer/interfaces: `^0.84.2`; it was resolved to `0.83.1`
	* @glimmer/reference: `^0.84.2`; it was resolved to `0.83.1`
```

A problem may be that I didn't install `@glimmer/interfaces` or `@glimmer/reference` myself, but the `0.83.1` versions were pulled in from `ember-template-lint`. I wonder if the specifications here are too strict and may have introduced a breaking change in a minor release.

What do you recommend that consumers of `ember-qunit` do?

Reply:
1. Note that they're *optional* peer dependencies: it's not a breaking change for someone to need to bump them when they *opt into* a new feature which requires it, but where it's otherwise not required to bump them.[^transitive]
2. Because the `@glimmer/*` packages are still 0.x, `0.83` → `0.84` was a breaking change, so it's not clear whether "widening" that range is safe.
3. Yes, you need to explicitly install the required version in `devDependencies` and specify that they have this same set of optional `peerDependencies` themselves (this is covered in [the migration notes](https://github.com/emberjs/ember-qunit/blob/master/docs/migration.md#migrating-to-native-typescript-support-in-v610)).

All that being said, I think there may be an issue with the 6.1 types until I'm able to publish `ember-resolver@8.1.0` on Monday, because of the `ember-qunit` → `@ember/test-helpers` → `ember-resolver` dependency chain.

Note: I wanted to answer this, but I won’t be around the rest of the weekend!

[^transitive]: It's therefore doubly reasonable for folks to add an explicit dependency rather than merely depending on a transitive in situations like this.

Reply:
FWIW this change is also breaking CI for crates.io :-/

- https://github.com/rust-lang/crates.io/pull/5679

Reply:
Thanks for the reports. Combined with the write-up and research I did over at #997, I concluded that this *is* breaking because `ember-quit` still supports Ember 3.28. Fixing this via #998; should have a 6.1.1 release out in a little with that.

Reply:
hmm, crates.io is on 4.x though, so I'm not sure why 3.28 compat would be related to this 🤔

Reply:
Ah, drat; I must be misremembering. I'll have to go check the version history for which VM version ended up on Ember v4 and see whether/how the compatibility changes for these specific types. (At some point we need to more firmly rationalize that whole set of relationships, too. 😩)

Reply:
v6.1.1 fixed it for now. thank you! :)

Reply:
Thanks for the report (to you as well, @ijlee2!).

Reply:
Thanks, @chriskrycho. I was able to install `ember-qunit@v6.1.1` in `ember-welcome-page` as well.